core/file.cpp

```diff
@@ -304,8 +341,13 @@ int parse_file(const char *filename, struct divelog *log)
 	}
 
 	/* Divesoft Freedom */
-	if (fmt && (!strcasecmp(fmt + 1, "DLF")))
-		return parse_dlf_buffer((unsigned char *)mem.data(), mem.size(), log);
+	if (fmt && (!strcasecmp(fmt + 1, "DLF"))) {
+		auto buffer = read_into_buffer(filename);
```

Comment:
We've already read the file once, just above:
auto [mem, err] = readfile(filename);

Why read it here again with another function into another format?